### PR TITLE
Alpine19 and Python venvs

### PR DESCRIPTION
**Pull request by jmthomas**
This isn't pressing, just an infrastructure improvement. Let's wait to merge until end of Oct / beginning of Nov with 5.20.0.

### CODE REVIEW

File: openc3-ruby/Dockerfile [hunk 1]
```diff
@@ -74,6 +74,8 @@ RUN apk update \
   && gem cleanup \
   && bundle config build.nokogiri --use-system-libraries \
   && bundle config git.allow_insecure true \
+  && python3 -m venv /openc3/venv \
+  && source /openc3/venv/bin/activate \
```
Comment (jmthomas) [inline openc3-ruby/Dockerfile:78]: Create the build time venv

File: openc3-cosmos-init/plugins/packages/openc3-cosmos-demo/targets/INST2/procedures/numpy.py [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+import numpy as np
+
+# Create a 2-D array, set every second element in
+# some rows and find max per row:
+x = np.arange(15, dtype=np.int64).reshape(3, 5)
+x[1:, ::2] = -99
+print(x)
+# array([[  0,   1,   2,   3,   4],
+#        [-99,   6, -99,   8, -99],
+#        [-99,  11, -99,  13, -99]])
```
Comment (jmthomas) [inline openc3-cosmos-init/plugins/packages/openc3-cosmos-demo/targets/INST2/procedures/numpy.py:11]: I thought this script was a good addition to the demo and it lets us ensure the Python package installation process is working

File: openc3-ruby/Dockerfile [hunk 3]
```diff
@@ -82,6 +84,9 @@ ENV IMAGE_USER=openc3
 ENV IMAGE_GROUP=openc3
 ENV USER_ID=1001
 ENV GROUP_ID=1001
+# The build time venv is at /openc3/venv and the runtime venv is at /gems/python_packages
+# NOTE: This is delared as PYTHONUSERBASE: "/gems/python_packages" in compose.yaml
+ENV PYTHONPATH=/openc3/venv/lib/python3.11/site-packages/:/gems/python_packages/lib/python3.11/site-packages/
```
Comment (jmthomas) [inline openc3-ruby/Dockerfile:89]: I was able to keep the existing `PYTHONUSERBASE` definition of the python packages doing this

**Comment by jmthomas** [comment]
How it shows up in the Packages tab:
Screenshot 2024-10-02 at 4 16 26 PM

**Review by ryanmelt — APPROVED** [review]

**Pull request merged**